index.js

```diff
@@ -1,12 +1,258 @@
+        if: {
+          allDay: false
+        }
+      },
+      endTime: {
+        label: 'End time',
+        type: 'time',
+        def: '05:30 PM',
+        required: true,
+        if: {
+          allDay: false
+        }
+      },
+      dateType: {
+        label: 'What type of event is this?',
+        help: 'Select if the event is on a single day, consecutive days, or repeats.',
+        type: 'select',
+        choices: [
+          {
+            label: 'Single Day',
+            value: 'single'
+          },
+          {
+            label: 'Consecutive Days',
+            value: 'consecutive'
+          },
+          {
+            label: 'Recurring',
+            value: 'repeat',
+          }
+        ],
+        def: 'single'
+      },
+      enddate: {
+        label: 'End date',
+        type: 'date',
+        if: {
+          dateType: 'consecutive'
+        }
+      },
+      repeatInterval: {
+        label: 'How often does the event repeat?',
+        type: 'select',
+        choices: [
+          {
+            label: 'Every week',
+            value: 'weeks'
+          },
+          {
+            label: 'Every month',
+            value: 'months'
+          }
+        ],
+        if: {
+          dateType: 'repeat'
+        }
+      },
+      repeatCount: {
+        label: 'How many times does it repeat?',
+        type: 'integer',
+        def: 1,
+        if: {
+          dateType: 'repeat'
+        }
+      },
+      description: {
+        type: 'string',
+        label: 'Description',
+        textarea: true,
+        required: true
+      },
+      image: {
+        label: 'Headline photo',
+        type: 'area',
+        options: {
+          max: 1,
+          widgets: {
+            '@apostrophecms/image': {}
+          }
+        },
+        required: false
+      }
+    },
+    group: {
+      basics: {
+        label: 'Basics',
+        fields: ['title', 'slug', 'description', 'startDate', 'allDay', 'startTime', 'endTime']
+      },
+      advanced: {
+        label: 'Advanced',
+        fields: ['dateType', 'enddate', 'repeatInterval', 'repeatCount']
+      },
+      meta: {
+        label: 'Meta',
+        fields: ['tags', 'published']
+      }
+    }
+  },
+
+  handlers(self, options) {
+    return {
+      beforeSave: {
+        async beforeSaveHandler(req, piece, options) {
+          self.denormalizeDatesAndTimes(piece)
+        }
+      },
+      afterInsert: {
+        async afterInsertHandler(req, piece, options) {
+          if (self._workflowPropagating) {
+            // if (self.isClone) {
+            // Workflow is replicating this but also its existing
+            // scheduled repetitions, don't re-replicate them and cause problems
+            // Don't allow replication of an event that is already a clone
+            console.log('Already replicating')
+            return
+          }
+          if (piece.dateType === 'repeat') {
+            await self.repeatEvent(req, piece, options)
```

Comment:
First make sure `piece.aposMode` is draft so that publishing doesn't also insert the duplicate events in the published mode.